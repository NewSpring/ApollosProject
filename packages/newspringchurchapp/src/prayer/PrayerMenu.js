import React, { PureComponent } from 'react';
import { Dimensions, View, Animated } from 'react-native';
import { SafeAreaView } from 'react-navigation';
import PropTypes from 'prop-types';
import { TabView } from 'react-native-tab-view';
import {
  TouchableScale,
  HorizontalTileFeed,
  PaddedView,
  styled,
  BodyText,
  H3,
} from '@apollosproject/ui-kit';
import AddPrayerCard from './AddPrayer/AddPrayerCard';
import PrayerMenuCard from './PrayerMenuCard';
import UserPrayerList from './UserPrayerList';
import PrayerPreviewCard from './PrayerPreviewCard';
import SavedPrayerList from './SavedPrayerList';

// used by PrayerList.js to determine what list shows when "Start Praying" is clicked
const prayerLists = {
  'my-church': 'ChurchPrayerList',
  'my-campus': 'CampusPrayerList',
  'my-community': 'GroupPrayerList',
};

// TODO: prayer preview cards should use the first one in the list
const getCategoryComponent = (key) => {
  switch (key) {
    case 'my-saved-prayers':
      return <SavedPrayerList />;
    case 'my-church':
      return (
        <PrayerPreviewCard
          avatarSource={{
            uri: 'https://rock.newspring.cc/GetImage.ashx?id=564401',
          }}
          avatarSize={'medium'}
          name={'Dan'}
          overlayColor={'#FFF'}
          prayer={
            'Pray that our church becomes activated and that we allow the Holy Spirit to drive what we do.'
          }
          source={'Anderson'}
          route={prayerLists[key]}
        />
      );
    case 'my-campus':
      return (
        <PrayerPreviewCard
          avatarSource={{
            uri: 'https://rock.newspring.cc/GetImage.ashx?id=564499',
          }}
          avatarSize={'medium'}
          name={'Morgan'}
          overlayColor={'#FFF'}
          prayer={
            'Pray that God will do big things at our Connect table this month.'
          }
          source={'Anderson'}
          route={prayerLists[key]}
        />
      );
    case 'my-community':
      return (
        <PrayerPreviewCard
          avatarSource={{
            uri: 'https://rock.newspring.cc/GetImage.ashx?id=576112',
          }}
          avatarSize={'medium'}
          name={'Devin'}
          overlayColor={'#FFF'}
          prayer={
            'Please pray for Fuse! FSKO is coming and we want God to show up big.'
          }
          source={'Greenwood'}
          route={prayerLists[key]}
        />
      );
    case 'my-prayers':
      return <UserPrayerList />;
    default:
      return null;
  }
};

const StyledFeed = styled(({ theme }) => ({
  paddingLeft: theme.sizing.baseUnit,
}))(HorizontalTileFeed);

const StyledPaddedView = styled(({ theme }) => ({
  marginTop: theme.sizing.baseUnit,
}))(PaddedView);

const StyledView = styled({
  height: Dimensions.get('window').height * 0.4,
  justifyContent: 'flex-end',
})(View);

const Tab = ({ description, component, showAddPrayerCard }) => (
  <>
    <StyledPaddedView>
      <BodyText>{description}</BodyText>
    </StyledPaddedView>
    {!showAddPrayerCard ? <StyledView>{component}</StyledView> : null}
  </>
);

Tab.propTypes = {
  description: PropTypes.string,
  component: PropTypes.func,
  showAddPrayerCard: PropTypes.bool,
};

class PrayerMenu extends PureComponent {
  static propTypes = {
    categories: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.string,
        description: PropTypes.string,
        image: PropTypes.string,
        overlayColor: PropTypes.arrayOf(PropTypes.string),
        title: PropTypes.string,
        key: PropTypes.string,
      })
    ),
  };

  static defaultProps = {
    categories: [],
  };

  state = {
    index: 0,
    routes: this.props.categories.map((category) => ({
      key: category.key,
    })),
    prayerMenuItemSelected: 1,
    showAddPrayerCard: true,
    animatedValue: new Animated.Value(0),
  };

  handleIndexChange = (index) => this.setState({ index });

  animate = (toValue) =>
    Animated.spring(this.state.animatedValue, {
      toValue,
      useNativeDriver: true,
      tension: 30,
      friction: 7,
    }).start();

  render() {
    return (
      <SafeAreaView>
        <AddPrayerCard
          description={
            'Take a moment to send a prayer request that your NewSpring Church family can pray for.'
          }
          {...this.props}
        />
        <PaddedView>
          <H3>Pray for Others</H3>
        </PaddedView>
        <TabView
          initialLayout={{
            height: Dimensions.get('window').height,
            width: Dimensions.get('window').width,
          }}
          navigationState={{ ...this.state }}
          renderScene={({ route }) => {
            const scenes = {};
            this.props.categories.forEach((category) => {
              scenes[category.key] = (
                <Tab
                  description={category.description}
                  showAddPrayerCard={this.state.showAddPrayerCard}
                  component={getCategoryComponent(category.key)}
                />
              );
            });
            return scenes[route.key];
          }}
          renderTabBar={(props) => (
            <StyledFeed
              content={this.props.categories}
              renderItem={({ item }) => (
                <TouchableScale
                  key={item.key}
                  onPress={() => {
                    if (this.state.showAddPrayerCard) {
                      this.animate(1);
                      this.setState({
                        showAddPrayerCard: false,
                      });
                    }
                    this.setState({ prayerMenuItemSelected: item.key });
                    props.jumpTo(item.key);
                  }}
                >
                  <PrayerMenuCard
                    image={item.image}
                    overlayColor={item.overlayColor}
                    title={item.title}
                    selected={this.state.prayerMenuItemSelected === item.key}
                  />
                </TouchableScale>
              )}
              loadingStateObject={{
                node: {
                  id: 'fakeId0',
                  isLoading: true,
                },
              }}
            />
          )}
          onIndexChange={this.handleIndexChange}
          swipeEnabled={false}
        />
      </SafeAreaView>
    );
  }
}

export default PrayerMenu;
