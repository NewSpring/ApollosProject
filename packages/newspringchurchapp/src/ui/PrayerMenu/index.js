import React, { PureComponent } from 'react';
import { View, Dimensions, Text } from 'react-native';
import PropTypes from 'prop-types';
import gql from 'graphql-tag';
import { Query } from 'react-apollo';
import {
  H3,
  HorizontalTileFeed,
  styled,
  TouchableScale,
  PaddedView,
  BodyText,
} from '@apollosproject/ui-kit';
import { TabView, SceneMap } from 'react-native-tab-view';
import PrayerMenuCard from '../PrayerMenuCard';
import UserPrayerList from '../UserPrayer';
import PrayerPreviewCard from '../../prayer/PrayerPreviewCard';
import AddPrayerCard from '../../prayer/AddPrayer/AddPrayerCard';

const RowHeader = styled(() => ({
  zIndex: 2, // UX hack to improve tapability. Positions RowHeader above StyledHorizontalTileFeed
}))(PaddedView);

const loadingStateObject = {
  node: {
    id: 'fakeId0',
    title: '',
    isLoading: true,
  },
};
// TODO: remove once this is pulling data
const PrayerPreviewCardComponent = () => (
  <PrayerPreviewCard
    imageSource={{
      uri: 'https://fillmurray.com/400/600',
    }}
    name={'Bill'}
    overlayColor={['#FFF', '#FFF']}
    prayer={
      'I’m alright. Nobody worry ’bout me. Why you got to gimme a fight? Can’t you just let it be?'
    }
    source={'Anderson'}
  />
);
// TODO: remove once this is pulling data
const StyledPrayerPreviewCardComponent = styled({
  position: 'absolute',
  bottom: 0,
  left: 0,
  right: 0,
})(PrayerPreviewCardComponent);

// TODO: this needs to be dynamic at some point
const prayerMenuData = [
  {
    id: '1',
    description: 'Pray for the prayers you haved saved',
    image: 'https://picsum.photos/600/400/?random',
    overlayColor: ['#6BAC43', '#6BAC43'],
    title: 'My Saved Prayers',
    key: 'saved',
  },
  {
    id: '2',
    description: 'Pray for the people in our church',
    image: 'https://picsum.photos/600/400/?random',
    overlayColor: ['#6BAC43', '#6BAC43'],
    title: 'My Church',
    key: 'church',
    component: <StyledPrayerPreviewCardComponent />,
  },
  {
    id: '3',
    description: 'Pray for the people at your campus',
    image: 'https://picsum.photos/600/400/?random',
    overlayColor: ['#6BAC43', '#6BAC43'],
    title: 'My Campus',
    key: 'campus',
    component: <PrayerPreviewCardComponent />,
  },
  {
    id: '4',
    description: 'Pray for those people in your community',
    image: 'https://picsum.photos/600/400/?random',
    overlayColor: ['#6BAC43', '#6BAC43'],
    title: 'My Community',
    key: 'community',
    component: <PrayerPreviewCardComponent />,
  },
  {
    id: '5',
    description: 'Revisit your prayers',
    image: 'https://picsum.photos/600/400/?random',
    overlayColor: ['#6BAC43', '#6BAC43'],
    title: 'My Prayers',
    key: 'prayers',
    component: <UserPrayerList />,
  },
];

const StyledFeed = styled(({ theme }) => ({
  paddingLeft: theme.sizing.baseUnit,
}))(HorizontalTileFeed);

const StyledPaddedView = styled(({ theme }) => ({
  height: Dimensions.get('window').height * 0.5,
  marginTop: theme.sizing.baseUnit * 2,
}))(PaddedView);

const StyledView = styled(() => ({
  height: '100%',
  justifyContent: 'flex-end',
}))(View);

const Tab = ({ index }) => {
  const data = prayerMenuData[index - 1];
  return (
    <StyledPaddedView>
      <BodyText>{data.description}</BodyText>
      <StyledView>{data.component}</StyledView>
    </StyledPaddedView>
  );
};
Tab.propTypes = {
  index: PropTypes.number,
};

class PrayerMenu extends PureComponent {
  state = {
    index: 0,
    routes: [
      {
        title: 'My Saved Prayers',
        key: 'saved',
      },
      {
        title: 'My Church',
        key: 'church',
      },
      {
        title: 'My Campus',
        key: 'campus',
      },
      {
        title: 'My Community',
        key: 'community',
      },
      {
        title: 'My Prayers',
        key: 'prayers',
      },
    ],
  };

  tabRoute = (index) => () => <Tab index={index} />;

  handleIndexChange = (index) => this.setState({ index });

  renderTabBar = (props) => (
    <StyledFeed
      content={prayerMenuData}
      renderItem={({ item }) => (
        <TouchableScale key={item.key} onPress={() => props.jumpTo(item.key)}>
          <PrayerMenuCard
            image={item.image}
            overlayColor={item.overlayColor}
            title={item.title}
          />
        </TouchableScale>
      )}
      loadingStateObject={loadingStateObject}
    />
  );

  render() {
    return (
      <>
        <Query
          query={gql`
            query {
              currentUser {
                id
                profile {
                  photo {
                    uri
                  }
                }
              }
            }
          `}
          fetchPolicy={'cache-and-network'}
        >
          {({ data }) => (
            <AddPrayerCard
              imgSrc={data.currentUser.profile.photo}
              {...this.props}
            />
          )}
        </Query>
        <RowHeader>
          <H3>Pray for Others</H3>
        </RowHeader>
        <TabView
          initialLayout={{
            height: Dimensions.get('window').height,
            width: Dimensions.get('window').width,
          }}
          navigationState={{ ...this.state }}
          renderScene={SceneMap({
            saved: this.tabRoute(1),
            church: this.tabRoute(2),
            campus: this.tabRoute(3),
            community: this.tabRoute(4),
            prayers: this.tabRoute(5),
          })}
          renderTabBar={this.renderTabBar}
          onIndexChange={this.handleIndexChange}
        />
      </>
    );
  }
}

export default PrayerMenu;
