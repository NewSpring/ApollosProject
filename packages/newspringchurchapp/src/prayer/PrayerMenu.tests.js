import React from 'react';
import renderer from 'react-test-renderer';

import Providers from 'newspringchurchapp/src/Providers';
import PrayerMenu from '.';

it('renders prayer without crashing', () => {
  const rendered = renderer
    .create(
      <Providers>
        <PrayerMenu />
      </Providers>
    )
    .toJSON();
  expect(rendered).toBeTruthy();
});
