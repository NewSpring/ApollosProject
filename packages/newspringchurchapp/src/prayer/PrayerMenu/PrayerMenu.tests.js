import React from 'react';
import renderer from 'react-test-renderer';

import Providers from 'newspringchurchapp/src/Providers';

import PrayerMenu from '.';

describe('the PrayerMenu component', () => {
  it('renders the prayer menu', () => {
    const tree = renderer.create(
      <Providers>
        <PrayerMenu />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
