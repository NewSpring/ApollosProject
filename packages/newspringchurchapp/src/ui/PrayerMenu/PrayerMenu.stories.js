import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { PaddedView } from '@apollosproject/ui-kit';

import PrayerMenu from '.';

storiesOf('PrayerMenu', module).add('Example', () => (
  <PaddedView>
    <PrayerMenu />
  </PaddedView>
));
