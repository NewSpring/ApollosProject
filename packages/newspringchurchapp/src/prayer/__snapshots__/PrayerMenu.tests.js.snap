// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders prayer menu 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <RCTSafeAreaView
      emulateUnlessSupported={true}
    >
      <View
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderBottomLeftRadius": 16,
            "flex": 1,
            "paddingHorizontal": 16,
            "paddingVertical": 64,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "paddingHorizontal": 16,
              "paddingVertical": 16,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "stretch",
                "flexDirection": "row",
                "flexWrap": "nowrap",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#303030",
                  "fontFamily": "Colfax-Black",
                  "fontSize": 24,
                  "lineHeight": 27.6,
                }
              }
            >
              Ask for prayer
            </Text>
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                bordered={false}
                disabled={false}
                pill={false}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#dddddd",
                    "borderColor": "#dddddd",
                    "borderRadius": 16,
                    "borderWidth": 2,
                    "elevation": 2,
                    "flexDirection": "row",
                    "height": 32,
                    "justifyContent": "center",
                    "marginBottom": 8,
                    "marginRight": 8,
                    "opacity": 1,
                    "overflow": "hidden",
                    "paddingHorizontal": 4,
                    "paddingVertical": 4,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "alignItems": "center",
                      "color": "#303030",
                      "fontFamily": "Colfax-Bold",
                      "fontSize": 12,
                      "justifyContent": "center",
                      "lineHeight": 17.28,
                      "paddingHorizontal": 6,
                      "textAlign": "center",
                    }
                  }
                >
                  My Prayers
                </Text>
              </View>
            </View>
          </View>
          <Text
            style={
              Object {
                "color": "#303030",
                "fontFamily": "System",
                "fontSize": 14,
                "fontStyle": null,
                "fontWeight": null,
                "lineHeight": 20.16,
                "paddingVertical": 16,
              }
            }
          >
            Take a moment to send a prayer request that your NewSpring Church family can pray for.
          </Text>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "padding": 16,
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              bordered={false}
              disabled={false}
              pill={true}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#6bac43",
                  "borderColor": "#6bac43",
                  "borderRadius": 48,
                  "borderWidth": 2,
                  "elevation": 2,
                  "flexDirection": "row",
                  "height": 48,
                  "justifyContent": "center",
                  "opacity": 1,
                  "overflow": "hidden",
                  "paddingHorizontal": 16,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "Colfax-Bold",
                    "fontSize": 16,
                    "lineHeight": 23.04,
                  }
                }
              >
                Add a Prayer Request
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "paddingHorizontal": 16,
            "paddingVertical": 16,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#303030",
              "fontFamily": "Colfax-Black",
              "fontSize": 24,
              "lineHeight": 27.6,
            }
          }
        >
          Pray for Others
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flex": 1,
              "overflow": "hidden",
            },
            undefined,
          ]
        }
      >
        <RCTScrollView
          contentContainerStyle={
            Object {
              "padding": 8,
              "paddingBottom": 0,
            }
          }
          data={Array []}
          decelerationRate="fast"
          disableVirtualization={false}
          getItem={[Function]}
          getItemCount={[Function]}
          horizontal={true}
          initialNumToRender={10}
          initialScrollIndex={0}
          keyExtractor={[Function]}
          loadingStateObject={
            Object {
              "node": Object {
                "id": "fakeId0",
                "isLoading": true,
              },
            }
          }
          maxToRenderPerBatch={10}
          numColumns={1}
          onContentSizeChange={[Function]}
          onEndReachedThreshold={2}
          onLayout={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          refreshing={false}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          showsHorizontalScrollIndicator={false}
          snapToAlignment="start"
          snapToInterval={512}
          stickyHeaderIndices={Array []}
          style={
            Object {
              "paddingLeft": 16,
            }
          }
          updateCellsBatchingPeriod={50}
          viewabilityConfigCallbackPairs={Array []}
          windowSize={21}
        >
          <View />
        </RCTScrollView>
        <View
          activeOffsetX={
            Array [
              -20,
              20,
            ]
          }
          enabled={false}
          failOffsetY={
            Array [
              -20,
              20,
            ]
          }
          onGestureEvent={[Function]}
          onHandlerStateChange={[Function]}
        >
          <View
            removeClippedSubviews={false}
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "transform": Array [
                  Object {
                    "translateX": undefined,
                  },
                ],
                "width": 0,
              }
            }
          />
        </View>
      </View>
    </RCTSafeAreaView>
  </View>
</RCTSafeAreaView>
`;

exports[`renders prayer menu with categories 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <RCTSafeAreaView
      emulateUnlessSupported={true}
    >
      <View
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderBottomLeftRadius": 16,
            "flex": 1,
            "paddingHorizontal": 16,
            "paddingVertical": 64,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "paddingHorizontal": 16,
              "paddingVertical": 16,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "stretch",
                "flexDirection": "row",
                "flexWrap": "nowrap",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#303030",
                  "fontFamily": "Colfax-Black",
                  "fontSize": 24,
                  "lineHeight": 27.6,
                }
              }
            >
              Ask for prayer
            </Text>
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                bordered={false}
                disabled={false}
                pill={false}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#dddddd",
                    "borderColor": "#dddddd",
                    "borderRadius": 16,
                    "borderWidth": 2,
                    "elevation": 2,
                    "flexDirection": "row",
                    "height": 32,
                    "justifyContent": "center",
                    "marginBottom": 8,
                    "marginRight": 8,
                    "opacity": 1,
                    "overflow": "hidden",
                    "paddingHorizontal": 4,
                    "paddingVertical": 4,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "alignItems": "center",
                      "color": "#303030",
                      "fontFamily": "Colfax-Bold",
                      "fontSize": 12,
                      "justifyContent": "center",
                      "lineHeight": 17.28,
                      "paddingHorizontal": 6,
                      "textAlign": "center",
                    }
                  }
                >
                  My Prayers
                </Text>
              </View>
            </View>
          </View>
          <Text
            style={
              Object {
                "color": "#303030",
                "fontFamily": "System",
                "fontSize": 14,
                "fontStyle": null,
                "fontWeight": null,
                "lineHeight": 20.16,
                "paddingVertical": 16,
              }
            }
          >
            Take a moment to send a prayer request that your NewSpring Church family can pray for.
          </Text>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "padding": 16,
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              bordered={false}
              disabled={false}
              pill={true}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#6bac43",
                  "borderColor": "#6bac43",
                  "borderRadius": 48,
                  "borderWidth": 2,
                  "elevation": 2,
                  "flexDirection": "row",
                  "height": 48,
                  "justifyContent": "center",
                  "opacity": 1,
                  "overflow": "hidden",
                  "paddingHorizontal": 16,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "Colfax-Bold",
                    "fontSize": 16,
                    "lineHeight": 23.04,
                  }
                }
              >
                Add a Prayer Request
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "paddingHorizontal": 16,
            "paddingVertical": 16,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#303030",
              "fontFamily": "Colfax-Black",
              "fontSize": 24,
              "lineHeight": 27.6,
            }
          }
        >
          Pray for Others
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flex": 1,
              "overflow": "hidden",
            },
            undefined,
          ]
        }
      >
        <RCTScrollView
          contentContainerStyle={
            Object {
              "padding": 8,
              "paddingBottom": 0,
            }
          }
          data={
            Array [
              Object {
                "description": "this is category 1",
                "id": "cat1",
                "image": "https://picsum.photos/300/300",
                "key": "cat1",
                "overlaycolor": Array [
                  "#fff",
                  "#fff",
                ],
                "title": "category 1",
              },
              Object {
                "description": "this is category 2",
                "id": "cat2",
                "image": "https://picsum.photos/300/300",
                "key": "cat2",
                "overlaycolor": Array [
                  "#fff",
                  "#fff",
                ],
                "title": "category 2",
              },
            ]
          }
          decelerationRate="fast"
          disableVirtualization={false}
          getItem={[Function]}
          getItemCount={[Function]}
          horizontal={true}
          initialNumToRender={10}
          initialScrollIndex={0}
          keyExtractor={[Function]}
          loadingStateObject={
            Object {
              "node": Object {
                "id": "fakeId0",
                "isLoading": true,
              },
            }
          }
          maxToRenderPerBatch={10}
          numColumns={1}
          onContentSizeChange={[Function]}
          onEndReachedThreshold={2}
          onLayout={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          refreshing={false}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          showsHorizontalScrollIndicator={false}
          snapToAlignment="start"
          snapToInterval={512}
          stickyHeaderIndices={Array []}
          style={
            Object {
              "paddingLeft": 16,
            }
          }
          updateCellsBatchingPeriod={50}
          viewabilityConfigCallbackPairs={Array []}
          windowSize={21}
        >
          <View>
            <View
              onLayout={[Function]}
              style={
                Array [
                  Object {
                    "flexDirection": "row",
                  },
                  null,
                ]
              }
            >
              <View
                accessible={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "transform": Array [
                      Object {
                        "scale": 1,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Object {
                      "height": 120,
                      "marginBottom": 8,
                      "marginRight": 12,
                      "marginTop": 8,
                      "transform": Array [
                        Object {
                          "scaleX": 0.95,
                        },
                        Object {
                          "scaleY": 1,
                        },
                      ],
                    }
                  }
                  transform={
                    Array [
                      Object {
                        "scaleX": 0.95,
                      },
                      Object {
                        "scaleY": 1,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "aspectRatio": 1,
                        "borderRadius": 16,
                        "height": "100%",
                        "overflow": "hidden",
                        "width": "100%",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "aspectRatio": 1,
                          "backgroundColor": "#dddddd",
                          "overflow": "hidden",
                          "width": "100%",
                        }
                      }
                    >
                      <View
                        animate={null}
                        customAnimate={null}
                        onReady={false}
                        style={
                          Object {
                            "aspectRatio": 1,
                            "backgroundColor": "#dddddd",
                            "width": "100%",
                          }
                        }
                      >
                        <Image
                          onLoad={[Function]}
                          source={
                            Array [
                              Object {
                                "cache": "force-cache",
                                "uri": "https://picsum.photos/300/300",
                                "url": "https://picsum.photos/300/300",
                              },
                            ]
                          }
                          style={
                            Object {
                              "aspectRatio": 1,
                              "backgroundColor": "#dddddd",
                              "opacity": 0,
                              "resizeMode": "cover",
                              "width": "100%",
                            }
                          }
                        />
                      </View>
                      <BVLinearGradient
                        colors={
                          Array [
                            3439329279,
                            3439329279,
                          ]
                        }
                        endPoint={
                          Object {
                            "x": 0,
                            "y": 1,
                          }
                        }
                        locations={
                          Array [
                            0.5,
                            1,
                          ]
                        }
                        startPoint={
                          Object {
                            "x": 0,
                            "y": 0,
                          }
                        }
                        style={
                          Object {
                            "bottom": 0,
                            "left": 0,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                          }
                        }
                      />
                    </View>
                    <Text
                      style={
                        Object {
                          "backgroundColor": "transparent",
                          "bottom": 16,
                          "color": "#ffffff",
                          "fontFamily": "Colfax-Medium",
                          "fontSize": 14,
                          "left": 16,
                          "lineHeight": 20.16,
                          "position": "absolute",
                          "right": 16,
                        }
                      }
                    >
                      category 1
                    </Text>
                  </View>
                </View>
              </View>
            </View>
            <View
              onLayout={[Function]}
              style={
                Array [
                  Object {
                    "flexDirection": "row",
                  },
                  null,
                ]
              }
            >
              <View
                accessible={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "transform": Array [
                      Object {
                        "scale": 1,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Object {
                      "height": 120,
                      "marginBottom": 8,
                      "marginRight": 12,
                      "marginTop": 8,
                      "transform": Array [
                        Object {
                          "scaleX": 0.95,
                        },
                        Object {
                          "scaleY": 1,
                        },
                      ],
                    }
                  }
                  transform={
                    Array [
                      Object {
                        "scaleX": 0.95,
                      },
                      Object {
                        "scaleY": 1,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "aspectRatio": 1,
                        "borderRadius": 16,
                        "height": "100%",
                        "overflow": "hidden",
                        "width": "100%",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "aspectRatio": 1,
                          "backgroundColor": "#dddddd",
                          "overflow": "hidden",
                          "width": "100%",
                        }
                      }
                    >
                      <View
                        animate={null}
                        customAnimate={null}
                        onReady={false}
                        style={
                          Object {
                            "aspectRatio": 1,
                            "backgroundColor": "#dddddd",
                            "width": "100%",
                          }
                        }
                      >
                        <Image
                          onLoad={[Function]}
                          source={
                            Array [
                              Object {
                                "cache": "force-cache",
                                "uri": "https://picsum.photos/300/300",
                                "url": "https://picsum.photos/300/300",
                              },
                            ]
                          }
                          style={
                            Object {
                              "aspectRatio": 1,
                              "backgroundColor": "#dddddd",
                              "opacity": 0,
                              "resizeMode": "cover",
                              "width": "100%",
                            }
                          }
                        />
                      </View>
                      <BVLinearGradient
                        colors={
                          Array [
                            3439329279,
                            3439329279,
                          ]
                        }
                        endPoint={
                          Object {
                            "x": 0,
                            "y": 1,
                          }
                        }
                        locations={
                          Array [
                            0.5,
                            1,
                          ]
                        }
                        startPoint={
                          Object {
                            "x": 0,
                            "y": 0,
                          }
                        }
                        style={
                          Object {
                            "bottom": 0,
                            "left": 0,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                          }
                        }
                      />
                    </View>
                    <Text
                      style={
                        Object {
                          "backgroundColor": "transparent",
                          "bottom": 16,
                          "color": "#ffffff",
                          "fontFamily": "Colfax-Medium",
                          "fontSize": 14,
                          "left": 16,
                          "lineHeight": 20.16,
                          "position": "absolute",
                          "right": 16,
                        }
                      }
                    >
                      category 2
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
        <View
          activeOffsetX={
            Array [
              -20,
              20,
            ]
          }
          enabled={false}
          failOffsetY={
            Array [
              -20,
              20,
            ]
          }
          onGestureEvent={[Function]}
          onHandlerStateChange={[Function]}
        >
          <View
            removeClippedSubviews={false}
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "transform": Array [
                  Object {
                    "translateX": undefined,
                  },
                ],
                "width": 1500,
              }
            }
          >
            <View
              accessibilityElementsHidden={false}
              importantForAccessibility="auto"
              style={
                Array [
                  Object {
                    "flex": 1,
                    "overflow": "hidden",
                  },
                  Object {
                    "width": 750,
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  Object {
                    "marginTop": 16,
                    "paddingHorizontal": 16,
                    "paddingVertical": 16,
                  }
                }
              >
                <Text
                  bold={false}
                  italic={false}
                  style={
                    Object {
                      "color": "#303030",
                      "fontFamily": "Colfax-Regular",
                      "fontSize": 16,
                      "lineHeight": 26,
                    }
                  }
                >
                  this is category 1
                </Text>
              </View>
            </View>
            <View
              accessibilityElementsHidden={true}
              importantForAccessibility="no-hide-descendants"
              style={
                Array [
                  Object {
                    "flex": 1,
                    "overflow": "hidden",
                  },
                  Object {
                    "width": 750,
                  },
                  undefined,
                ]
              }
            />
          </View>
        </View>
      </View>
    </RCTSafeAreaView>
  </View>
</RCTSafeAreaView>
`;
